### ISSUE TITLE
Project status

### ISSUE THREAD

**Issue by XXFOGS** [quarkslab/mattermost-plugin-e2ee#42]
Has this project been abondoned? No changes have been made for the last year.

**Comment by pive** [comment]
Hey @XXFOGS, it's not abandoned. The mattermost app has undergone many changes to newer versions, and we are currently looking into the possibility of an update to support these versions.

**Comment by XXFOGS** [comment]
Hi @pive, good to know, is this project planned to be maintained by quarkslab for as long as they employ mattermost or otherwise extended period of time?

**Comment by pive** [comment]
yes, we use the plugin therefore we want to maintain it for as long as we use it but the lead developer is currently  no longer at Quarkslab so we are figuring out who can take the lead.

**Comment by colinstu12** [comment]
Any update/status on project status / it's been 8mo? Curious, thanks!

**Comment by Kuuchuu** [comment]
Would also love for this plugin to get an update for the latest version of Mattermost

**Comment by joltcoke** [comment]
I've created a series of PR's to get the plugin up and running again. We using it on our production installation for a while now. If someone is willing to accept the changes into mainline I'm happy to open pull requests against this repo. If someone wants to try it, feel free to grab the build from this action run:

https://github.com/joltcoke/mattermost-plugin-e2ee/actions/runs/32719107613#artifacts



### Foundation (1–2)
                                                                                                                                                                                          
  **[#1 — Modernize dependencies (drop archived/deprecated packages)](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/1)**
  `webapp-modernize-deps` → `main` · CI passing

  Swapped `node-webcrypto-ossl` for Node's native WebCrypto and `node-sass` for Dart Sass — both archived/deprecated upstream. Kickoff of the whole modernization effort.

  **[#2 — Fix version check gating message-update hooks](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/2)**
  `fix-message-edit-version-check` → `webapp-modernize-deps` · CI passing

  Real production bug: edited messages appeared unencrypted and un-flagged on Mattermost ≥6.6, because the version-gate only checked `major == 6`. Fixed to handle `major > 6` too.

  ---                                                                                                                                                                                     

  ### Webapp build tooling (3–7)
                                                                                                                                                                                          
  **[#3 — Upgrade webpack to 5](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/3)**
  `upgrade-webpack` → `fix-message-edit-version-check` · CI passing

  Devtool/externals syntax updates for webpack 5; eliminated the OpenSSL3/MD4 workaround webpack 4 needed on modern Node.

  **[#4 — Upgrade TypeScript to 5.9.3](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/4)**
  `upgrade-typescript` → `upgrade-webpack` · CI passing

  3.9 → 5.9.3. Surfaced a real gap: `react-intl` was never actually installed, only stale type definitions existed.
  
  **[#5 — Upgrade openpgp to 6.3.1](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/5)**                                                                                          
  `upgrade-openpgp` → `upgrade-typescript` · CI passing

  Migrated the openpgp.js 4→6 API (`readKeys`, `createMessage`, `encryptionKeys`) throughout the GPG backup/restore code.

  **[#6 — Upgrade babel-loader to 10.1.1](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/6)**
  `upgrade-babel-loader` → `upgrade-openpgp` · CI passing

  Minimal, scoped bump — the companion piece webpack 5 needed to fully drop the OpenSSL3 workaround.

  **[#7 — Upgrade jest to 30.4.2](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/7)**                                                                                            
  `upgrade-jest` → `upgrade-babel-loader` · CI passing

  26 → 30. Needed a custom jsdom environment subclass for `structuredClone`, plus resolver and export-condition fixes for openpgp's ESM build.                                            

  ---
                                                                                                                                                                                          
  ### Go runtime & SDK (8–9)

  **[#8 — Upgrade to Go 1.26, migrate golangci-lint to v2](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/8)**
  `upgrade-go-deps` → `upgrade-jest` · CI passing

  golangci-lint v1 refuses to lint `go 1.26`-declared code, so migrated the whole config to the v2 schema. Deferred four `crypto/elliptic` deprecation findings with a documented rationale rather
  than a risky drive-by migration of security-sensitive code.

  **[#9 — Migrate SDK to mattermost/server/public](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/9)**
  `upgrade-mattermost-server-sdk` → `upgrade-go-deps` · CI passing

  The old `mattermost-server` module has no fetchable version past 2022's v6.7.2. Migrated to the actively-maintained monorepo module instead — found and hand-ported the removed `EnsureBot` helper,
  derived `min_server_version` from actual API usage rather than copying a reference template's pin.

  > → **govulncheck: 191 reachable findings → 0**

  ---                                                                                                                                                                                     

  ### Webapp ecosystem catch-up (10–13)
                                                                                                                                                                                          
  **[#10 — Upgrade eslint to 8, drop emotion](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/10)**
  `upgrade-eslint-emotion` → `upgrade-mattermost-server-sdk` · CI passing

  Adopted `@mattermost/eslint-plugin`'s shared config — not ESLint 9, after checking that the real, currently-maintained starter template is still on 8.x. Removed `@emotion`, unused anywhere in this
  plugin's own source.
  **[#11 — Upgrade react, react-redux and redux](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/11)**
  `upgrade-react-redux` → `upgrade-eslint-emotion` · CI passing

  Matched the *real* Mattermost webapp runtime (18.2.0 / 9.2.0 / 5.0.1) rather than npm's "latest" tag, since all three are webpack externals resolved from whatever the host page actually injects.

  **[#12 — Upgrade mattermost-redux, 5.33.1 → 11.9.0](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/12)**
  `upgrade-mattermost-redux` → `upgrade-react-redux` · CI passing

  Six major versions, and genuinely bundled (not an external) so the jump mattered for real. Domain types moved to `@mattermost/types`, `Client4` to `@mattermost/client`, and `ActionFunc` split into
  sync/async variants throughout the action layer.

  **[#13 — Bump misc, babel, react-intl, react-bootstrap types](https://github.com/joltcoke/mattermost-plugin-e2ee/pull/13)**
  `upgrade-misc-deps` → `upgrade-mattermost-redux` · CI passing

  Loaders, core-js, and type packages brought current, plus `react-intl` 6→10 (unblocked now that React is current). Investigated Babel 8 and backed off — `preset-env` dropped the `useBuiltIns`
  option this plugin relies on, and the current official template is still on 7.x too.
